api/src/main/java/org/openmrs/module/appframework/domain/Requireable.java

```diff
@@ -0,0 +1,13 @@
+package org.openmrs.module.appframework.domain;
+
+import org.openmrs.module.appframework.context.AppContextModel;
+
+/**
+ * An interface for objects which can be required or excluded by evaluating a Javascript expression using {@link org.openmrs.module.appframework.service.AppFrameworkService#checkRequireExpression(Requireable, AppContextModel)}
+ */
+public interface Requireable {
+
+	String getId();
+
+	String getRequire();
```

Comment:
How about getRequireExpression?

Reply:
I left this as `getRequire()` because that way it works with already-existing JSON documents that describe an `Extension`. This is basically just an interface that extracts the parts of `Extension` that are needed for the `checkRequireExpression()` method to function.

Reply:
Fair enough!